omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py

```diff
@@ -2395,26 +2624,24 @@ def paste_image(self, pil_img, img_name, panel, page, dpi=None):
         x = x - page['x']
         y = y - page['y']
 
+        x2 = scale_to_export_dpi(x + width)
+        y2 = scale_to_export_dpi(y + height)
         x = scale_to_export_dpi(x)
         y = scale_to_export_dpi(y)
-        width = scale_to_export_dpi(width)
-        height = scale_to_export_dpi(height)
-
-        x = int(round(x))
-        y = int(round(y))
-        width = int(round(width))
-        height = int(round(height))
+        width = x2 - x
+        height = y2 - y
```

Comment:
scaling the width/height independently can produce pixel offsets because of the rounding.
`round(x) + round(width) != round(x + width)`

But that created another problem that can change the ratio `width/height` (if `height == width` before rounding, this is not always true after the rounding because of the decimal values of x and y)
That's something I need to fix

Reply:
In the end, I think it's better to leave it this way. Square panels would have a difference of maximum one pixel between width and height.
The only way I thought of handling it would require updating the position of the panels beforehand, then make sure alignments remain, ...   too much code for something barely noticeable IMO